src/main/kotlin/org/cascadebot/cascadebot/music/MusicHandler.kt

```diff
@@ -0,0 +1,197 @@
+package org.cascadebot.cascadebot.music
+
+import com.google.gson.JsonParser
+import com.sedmelluq.discord.lavaplayer.player.AudioPlayer
+import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager
+import com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager
+import com.sedmelluq.discord.lavaplayer.source.AudioSourceManagers
+import com.sedmelluq.discord.lavaplayer.source.bandcamp.BandcampAudioSourceManager
+import com.sedmelluq.discord.lavaplayer.source.beam.BeamAudioSourceManager
+import com.sedmelluq.discord.lavaplayer.source.soundcloud.SoundCloudAudioSourceManager
+import com.sedmelluq.discord.lavaplayer.source.twitch.TwitchStreamAudioSourceManager
+import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager
+import lavalink.client.io.jda.JdaLavalink
+import net.dv8tion.jda.api.entities.Guild
+import net.dv8tion.jda.api.entities.TextChannel
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Request
+import okhttp3.Response
+import org.apache.commons.lang3.StringUtils
+import org.apache.http.client.config.CookieSpecs
+import org.apache.http.client.config.RequestConfig
+import org.cascadebot.cascadebot.CascadeBot
+import org.cascadebot.cascadebot.data.Config
+import org.cascadebot.cascadebot.data.language.Language
+import org.cascadebot.cascadebot.data.managers.GuildDataManager
+import org.cascadebot.cascadebot.data.objects.Flag
+import org.cascadebot.cascadebot.events.PlayerListener
+import org.cascadebot.cascadebot.messaging.MessageType
+import org.cascadebot.cascadebot.messaging.Messaging
+import org.cascadebot.cascadebot.messaging.Messaging.sendExceptionMessage
+import org.cascadebot.cascadebot.utils.PasteUtils
+import java.io.IOException
+import java.net.URI
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.util.ArrayList
+import java.util.function.Consumer
+import java.util.function.Function
+import java.util.regex.Pattern
+
+class MusicHandler {
+    val playerManager: AudioPlayerManager = DefaultAudioPlayerManager()
+    val typePattern = Pattern.compile("youtube#([A-z]+)")
+
+    val youtubeSourceName: String
+    val twitchSourceName: String
+
+    val players: MutableMap<Long, CascadePlayer> = HashMap()
+    var musicJsonParser = JsonParser()
+    lateinit var lavaLink: JdaLavalink
+        private set
```

Comment:
wat

Reply:
This is a variable that is late initialised and has a private setter. What's the problem?

Reply:
ahhhh. the `private set` looked like a variable